showcase/scripts/generate_showcase.sh

```diff
@@ -1,129 +1,95 @@
 #!/bin/bash
-# Generates showcase without post processing (i.e. raw gapic, grpc and proto
-# libraries). It will compute the showcase version from
-# `sdk-platform-java/showcase/gapic-showcase/pom.xml`. The generator version is
-# inferred from versions.txt
+# Generates the showcase library using the docker image, which is built
+# from the current state of the repo in order to test local changes.
 set -ex
 
-readonly SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
-readonly LIB_GEN_SCRIPTS_DIR="${SCRIPT_DIR}/../../hermetic_build/library_generation/"
-source "${LIB_GEN_SCRIPTS_DIR}/tests/test_utilities.sh"
-source "${LIB_GEN_SCRIPTS_DIR}/utils/utilities.sh"
-readonly perform_cleanup=$1
+readonly ROOT_DIR="$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )/../../"
+pushd "${ROOT_DIR}"
 
-cd "${SCRIPT_DIR}"
-mkdir -p "${SCRIPT_DIR}/output"
+while [[ $# -gt 0 ]]; do
+key="$1"
+case "${key}" in
+  --replace)
+    replace="$2"
+    shift
+    ;;
+  *)
+    echo "Invalid option: [$1]"
+    exit 1
+    ;;
+esac
+shift
+done
 
+if [ -z "${replace}" ]; then
+  replace="false"
+fi
+
+# download api definitions from googleapis repository
+googleapis_commitish=$(grep googleapis_commitish generation_config.yaml | cut -d ":" -f 2 | xargs)
+api_def_dir=$(mktemp -d)
+git clone https://github.com/googleapis/googleapis.git "${api_def_dir}"
+
+pushd "${api_def_dir}"
+git checkout "${googleapis_commitish}"
+# for local setups, we avoid permission issues when the docker image
+# performs version-dependent operations.
+rm -rf ".git/"
+popd
+
+# append showcase definitions to googleapis repository
+showcase_def_dir=$(mktemp -d)
+git clone https://github.com/googleapis/gapic-showcase.git "${showcase_def_dir}"
 get_version_from_pom() {
   target_pom="$1"
   key="$2"
   # prints the result to stdout
   grep -e "<${key}>" "${target_pom}" | cut -d'>' -f2 | cut -d'<' -f1
 }
+# looks at sdk-platform-java/showcase/gapic-showcase/pom.xml to extract the
+# version of gapic-showcase
+# see https://github.com/googleapis/gapic-showcase/releases
+showcase_version=$(get_version_from_pom \
```

Comment:
Is this used? We should still use it to checkout showcase repo with a release tag.

Reply:
It should be used, exactly for checking out at the required tag. I'll fix this. Thanks

Reply:
Done